src/analysis/normed_space/M_structure.lean

```diff
@@ -0,0 +1,362 @@
+      map_sub,continuous_linear_map.coe_mul]
+  ... = ∥Q (P (Q x))∥ + ∥(1-Q) (P (Q x))∥ + (∥Q x - Q (P (Q x))∥ + ∥(1-Q) (P (Q x))∥) :
+    by rw [sub_mul, ← sq, projection_def h₂.left, one_mul, sub_self,
+      continuous_linear_map.zero_apply, zero_sub, norm_neg]
+  ... = ∥Q (P (Q x))∥ + ∥Q x - Q (P (Q x))∥ + 2•∥(1-Q) (P (Q x))∥  : by abel
+  ... ≥ ∥Q x∥ + 2 • ∥ (P * Q) x - (Q * P * Q) x∥ :
+    by exact add_le_add_right (norm_le_insert' (Q x) (Q (P (Q x)))) (2•∥(1-Q) (P (Q x))∥),
+  rw ge at e1,
+  nth_rewrite_rhs 0 ← add_zero (∥Q x∥) at e1,
+  rw [add_le_add_iff_left, two_smul,  ← two_mul]  at e1,
+  rw le_antisymm_iff,
+  refine ⟨_, norm_nonneg _⟩,
+  rwa [←mul_zero (2:ℝ), mul_le_mul_left (show (0:ℝ) < 2, by norm_num)] at e1
+end
+
+lemma QP_eq_QPQ (P Q : X →L[𝕜] X) (h₁: is_Lprojection P) (h₂: is_Lprojection Q) : Q * P = Q * P * Q
+  :=
+begin
+  have e1: P * (1 - Q) = P * (1 - Q) - (Q * P - Q * P * Q) :=
+  calc P * (1 - Q) = (1 - Q) * P * (1 - Q) : by rw PQ_eq_QPQ P (1 - Q) h₁ h₂.Lcomplement
+  ... = P * (1 - Q) - (Q * P - Q * P * Q) :
+    by rw [mul_assoc, sub_mul, one_mul, mul_sub, mul_one, mul_sub Q, mul_assoc],
+  rwa [eq_sub_iff_add_eq, add_right_eq_self, sub_eq_zero] at e1
+end
+
+lemma Lproj_commute {P Q: X →L[𝕜] X} (h₁: is_Lprojection P) (h₂ : is_Lprojection Q) : commute P Q :=
+show P * Q = Q * P, by rw [QP_eq_QPQ P Q h₁ h₂, PQ_eq_QPQ P Q h₁ h₂]
+
+lemma mul {P Q : X →L[𝕜] X} (h₁ : is_Lprojection P) (h₂ : is_Lprojection Q) :
+  is_Lprojection (P*Q) :=
+begin
+  refine ⟨is_projection.mul_of_commute (Lproj_commute h₁ h₂) h₁.left h₂.left, _⟩,
+  intro x,
+  refine le_antisymm _ _,
+  { calc ∥ x ∥ = ∥(P * Q) x + (x - (P * Q) x)∥ : by abel
+    ... ≤ ∥(P * Q) x∥ + ∥ x - (P * Q) x ∥ : by apply norm_add_le
+    ... = ∥(P * Q) x∥ + ∥(1 - P * Q) x∥ : by rw [continuous_linear_map.sub_apply,
+      continuous_linear_map.one_apply] },
+  { calc ∥x∥ = ∥P(Q x)∥ + (∥Q x - P (Q x)∥ + ∥x - Q x∥) : by rw [h₂.right x, h₁.right (Q x),
+      continuous_linear_map.sub_apply, continuous_linear_map.one_apply,
+      continuous_linear_map.sub_apply, continuous_linear_map.one_apply, add_assoc]
+    ... ≥ ∥P(Q x)∥ + ∥(Q x - P (Q x)) + (x - Q x)∥ :
+      by apply (add_le_add_iff_left (∥P(Q x)∥)).mpr (norm_add_le (Q x - P (Q x)) (x - Q x))
+    ... = ∥(P * Q) x∥ + ∥(1 - P * Q) x∥ : by rw [sub_add_sub_cancel',
+      continuous_linear_map.sub_apply, continuous_linear_map.one_apply,
+      continuous_linear_map.coe_mul] }
+end
+
+lemma join {P Q: X →L[𝕜] X} (h₁ : is_Lprojection P) (h₂ : is_Lprojection Q) :
+  is_Lprojection (P + Q - P * Q) :=
+begin
+  have e1:  1 - (1 - P) * (1 - Q) = P + Q - P * Q,
+  { rw [sub_mul, one_mul, sub_sub, sub_sub_self, mul_sub, mul_one, add_sub, add_comm] },
+  rw [← e1, ← is_Lprojection.Lcomplement_iff],
+  exact is_Lprojection.mul (is_Lprojection.Lcomplement h₁) (is_Lprojection.Lcomplement h₂)
+end
+
+instance : has_compl { f : X →L[𝕜] X // is_Lprojection f } :=
+⟨λ P, ⟨1-P, P.prop.Lcomplement⟩⟩
```

Comment:
There are a bunch of small mathlib-style inaccuracies throughout, of which this is just one example.
```suggestion
⟨λ P, ⟨1 - P, P.prop.Lcomplement⟩⟩
```
The things I noticed in various places that should be as follows:

1. infix operators get a space between the operator and each argument
2. prefix and postfix notations (like ↑P or Pᶜ) do *not* get a space
3. type ascriptions get a space on either side of the colon (i.e., `1 : ℕ`, not `1:ℕ`).

For a complete description, see the [style guide](https://leanprover-community.github.io/contribute/style.html)

Reply:
@j-loreaux Hopefully I've fixed all of these now (including [one](https://github.com/leanprover-community/leanprover-community.github.io/pull/265) in the style guide itself. Please let me know if you spot any more.